src/sds_data_model/_table.py

```diff
@@ -0,0 +1,26 @@
+from pandas import to_numeric
+   
+# Updates datatypes to smaller memory allocation, except dtype int8 which is not accepted by rasterio package, so converted to int16
+def _update_datatypes(
+    df 
+):    
+    #identify float and integer columns, which are treated separately to other dtypes
+    fcols = df.select_dtypes('float').columns
+    icols = df.select_dtypes('integer').columns
+
+    df[fcols] = df[fcols].apply(to_numeric, downcast = 'float')
+    df[icols] = df[icols].apply(to_numeric, downcast = 'integer')
```

Comment:
Though this isn't an example of [chained indexing](https://pandas.pydata.org/docs/user_guide/indexing.html#returning-a-view-versus-a-copy), it's preferable to use the `.loc` method in assignment just in case.

```suggestion
    df.loc[:, fcols] = df.loc[:, fcols].apply(to_numeric, downcast = 'float')
    df.loc[:, icols] = df.loc[:, icols].apply(to_numeric, downcast = 'integer')
    df.loc[:, icols] = df.loc[:, icols].apply(_convert_int_dtype)
```

N.B. In this suggestion, I'm assuming that you've move `convert_int_dtype` out of the body of `_update_dtypes` and prefixed it with an underscore.